<?xml version="1.0" encoding="UTF-8"?>

<launch>

  <arg name="use_tf" default="true"/>

  <group ns="eagleye">

    <arg name="use_rtk_dead_reckoning" default="true"/>
    <arg name="use_rtk_heading" default="false"/>
    <arg name="enable_additional_rolling" default="false"/>
    <arg name="use_multi_antenna_mode" default="true"/>
    <arg name="yaml_file" default="$(find eagleye_rt)/config/eagleye_config.yaml"/>

    <param name="yaml_file" value="$(arg yaml_file)"/>
    <param name ="velocity_scale_factor_save_str" value="$(find eagleye_rt)/config/velocity_scale_factor.txt"/>
    <param name="use_multi_antenna_mode" value="$(arg use_multi_antenna_mode)"/>

    <rosparam command="load" file="$(find eagleye_rt)/config/eagleye_config.yaml"/>

    <node pkg="eagleye_rt" name="twist_relay" type="twist_relay" />
    <node pkg="eagleye_rt" name="tf_converted_imu" type="tf_converted_imu" />
    <node pkg="eagleye_rt" name="velocity_scale_factor_node" type="velocity_scale_factor" />
    <node pkg="eagleye_rt" name="yaw_rate_offset_stop_node" type="yaw_rate_offset_stop" />
    <node pkg="eagleye_rt" name="yaw_rate_offset_node_1st" type="yaw_rate_offset" args="1st"/>
    <node pkg="eagleye_rt" name="yaw_rate_offset_node_2nd" type="yaw_rate_offset" args="2nd"/>
    <node pkg="eagleye_rt" name="heading_node_1st" type="heading" args="1st" if="$(eval use_rtk_heading == false)"/>
    <node pkg="eagleye_rt" name="heading_node_2nd" type="heading" args="2nd" if="$(eval use_rtk_heading == false)"/>
    <node pkg="eagleye_rt" name="heading_node_3rd" type="heading" args="3rd" if="$(eval use_rtk_heading == false)"/>
    <node pkg="eagleye_rt" name="heading_interpolate_node_1st" type="heading_interpolate" args="1st"/>
    <node pkg="eagleye_rt" name="heading_interpolate_node_2nd" type="heading_interpolate" args="2nd"/>
    <node pkg="eagleye_rt" name="heading_interpolate_node_3rd" type="heading_interpolate" args="3rd"/>
    <node pkg="eagleye_rt" name="slip_angle_node" type="slip_angle" />
    <node pkg="eagleye_rt" name="distance_node" type="distance" />
    <node pkg="eagleye_rt" name="trajectory_node" type="trajectory" />
    <node pkg="eagleye_rt" name="position_node" type="position" />
    <node pkg="eagleye_rt" name="position_interpolate_node" type="position_interpolate"/>
    <node pkg="eagleye_rt" name="smoothing_node" type="smoothing"/>
    <!-- <node pkg="eagleye_rt" name="height_node" type="height"/> -->
    <node pkg="eagleye_rt" name="angular_velocity_offset_stop_node" type="angular_velocity_offset_stop"/>
    <node pkg="eagleye_rt" name="correction_imu" type="correction_imu"/>
    <node pkg="eagleye_rt" name="monitor" type="monitor"  output="screen"/>
    <node pkg="eagleye_rt" name="rolling" type="rolling" />

    <!-- RTK Options -->
    <node pkg="eagleye_rt" name="rtk_dead_reckoning" type="rtk_dead_reckoning" if="$(arg use_rtk_dead_reckoning)"/>
    <node pkg="eagleye_rt" name="rtk_heading_node_1st" type="rtk_heading" args="1st" if="$(arg use_rtk_heading)"/>
    <node pkg="eagleye_rt" name="rtk_heading_node_2nd" type="rtk_heading" args="2nd" if="$(arg use_rtk_heading)"/>
    <node pkg="eagleye_rt" name="rtk_heading_node_3rd" type="rtk_heading" args="3rd" if="$(arg use_rtk_heading)"/>
    <node pkg="eagleye_rt" name="enable_additional_rolling_node" type="enable_additional_rolling" if="$(arg enable_additional_rolling)"/>

    <!-- Multi Antennas Options -->
    <include file="$(find gnss_compass)/launch/gnss_compass.launch">
    </include>

  <include file="$(find eagleye_nmea2fix)/launch/nmea2fix_dualantenna.launch">
        <arg name="sub_topic_name1" default="/sensing/gnss/nmea_sentence"/>
        <arg name="output_gga" default="true"/>
        <arg name="output_rmc" default="true"/>
        <arg name="sub_topic_name2" default="/sensing/sub_gnss/nmea_sentence"/>
  </include>

  </group>

  <include file="$(find eagleye_tf)/launch/tf.launch" if="$(arg use_tf)"/>

  <include file="$(find eagleye_fix2pose)/launch/fix2pose.launch">
    <arg name="gnss_frame_id" default="gnss_link"/> 
  </include>

</launch>
